hopp/simulation/technologies/wind/pysam_financial.py

```diff
@@ -0,0 +1,26 @@
+# tools to add floris to the hybrid simulation class
+from attrs import define, field
+
+import PySAM.Singleowner as Singleowner
+
+from hopp.simulation.base import BaseClass
+from hopp.type_dec import resource_file_converter
+from hopp.utilities.utilities import load_yaml
+
+
+@define
+class PySAMFinancial(BaseClass):
+    config_dict: dict = field(converter=dict)
+
+    financial_model: Singleowner = field(init=False)
+
+    def __attrs_post_init__(self):
+        input_file_path = resource_file_converter(self.config_dict["financial_input_file"])
+        filetype = input_file_path.suffix.strip(".")
+        if filetype.lower() in ("yml", "yaml"):
+            input_dict = load_yaml(input_file_path)
+        else:
+            raise ValueError("Supported import filetype is YAML")
+            
+        self.financial_model = Singleowner.new()
```

Comment:
So originally when PySAM.Windpower is paired with PySAM.Singleowner, the Singleowner is created with `from_existing` instead of `new` so that the underlying data is shared between the Windpower and the Singleowner. 

https://github.com/NREL/HOPP/blob/fa6b5ab1979c28d50cb4e1b098997d563c0b25c2/hybrid/wind_source.py#L49C16-L49C16

Without that data sharing, such as for the CustomFinancialModel, there will need to be explicit functions for copying variables over: `set_financial_inputs`

https://github.com/NREL/HOPP/blob/fa6b5ab1979c28d50cb4e1b098997d563c0b25c2/hybrid/power_source.py#L59

Reply:
I think I need to discuss more to understand what we want to accomplish here.